dspace-api/src/main/java/org/dspace/storage/bitstore/AzureBlobStorageBitStoreService.java

```diff
@@ -0,0 +1,395 @@
+
+					log.debug("Attributes raw: " + attrs);
+
+					if (attrs != null) {
+
+						log.debug("Attributes as String: " + attrs.toString());
+
+						if (lenght != null && attrs.containsKey("size_bytes")) {
+							attrs.put("size_bytes", lenght);
+						}
+
+						if (checksum != null && attrs.containsKey("checksum")) {
+							attrs.put("checksum", checksum);
+							attrs.put("checksum_algorithm", this.algorithm);
+						}
+
+						if (lastModifiedTimeString != null && attrs.containsKey("modified")) {
+							attrs.put("modified", lastModifiedTimeString);
+						}
+
+						return attrs;
+					} else {
+						// not sure if this would ever happen
+						return null;
+					}
+				}
+			}
+
+		} catch (Exception e) {
+			log.error("about(" + key + ", attrs)", e);
+			throw new IOException(e);
+		}
+
+		return null;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.dspace.storage.bitstore.BitStoreService#remove(org.dspace.content.
+	 * Bitstream)
+	 */
+	@Override
+	public void remove(Bitstream bitstream) throws IOException {
+
+		String key = getFullKey(bitstream.getInternalId());
+		log.info("Attempting removal of Azure Blob storage file: " + key);
+
+		try {
+			CloudBlob blob = this.blobContainer.getBlockBlobReference(key);
+			if (blob.exists()) {
+				blob.delete();
+				if (!blob.exists()) {
+					log.info("Removal of Azure Blob storage file: " + key + " was successful");
+				}
+			} else {
+				log.warn("File didn't exist in Azure Blob Storage " + key);
+			}
+		} catch (Exception e) {
+			log.error("remove(" + key + ")", e);
+			throw new IOException(e);
+		}
+
+	}
+
+	/**
+	 * Creates and returns a temporary local file for use by the sample.
+	 *
+	 * @param blockBlob
+	 *            CloudBlockBlob object.
+	 * @param filePath
+	 *            The path to the file to be uploaded.
+	 *
+	 * @throws Throwable
+	 */
+	private void uploadFileBlocksAsBlockBlob(CloudBlockBlob blockBlob, String filePath) throws Throwable {
+
+		FileInputStream fileInputStream = null;
+		try {
+			// Open the file
+			fileInputStream = new FileInputStream(filePath);
+
+			// Split the file into 4mb blocks
+			int blockNum = 0;
+			String blockId = null;
+			String blockIdEncoded = null;
+			ArrayList<BlockEntry> blockList = new ArrayList<BlockEntry>();
+			while (fileInputStream.available() > (4 * 1024 * 1024)) {
+				blockId = String.format("%05d", blockNum);
+				blockIdEncoded = Base64.getEncoder().encodeToString(blockId.getBytes());
+				blockBlob.uploadBlock(blockIdEncoded, fileInputStream, (32 * 1024));
+				blockList.add(new BlockEntry(blockIdEncoded));
+				blockNum++;
+			}
+			blockId = String.format("%05d", blockNum);
+			blockIdEncoded = Base64.getEncoder().encodeToString(blockId.getBytes());
+			blockBlob.uploadBlock(blockIdEncoded, fileInputStream, fileInputStream.available());
+			blockList.add(new BlockEntry(blockIdEncoded));
+
+			// Commit the blocks
+			blockBlob.commitBlockList(blockList);
+		} catch (Throwable t) {
+			throw t;
```

Comment:
This `catch` doesn't seem to be needed.  It's just re-throwing the same exception.